### PR TITLE
Feature/profile

### PR DESCRIPTION
**Pull request by letrylesha**
image
image

### CODE REVIEW

File: src/components/App/ContentWrapper/HomePage/Tiles/Tiles.tsx [hunk 1]
```diff
@@ -28,7 +28,7 @@ const Tiles: React.FC<Props> = ({ tiles }) => {
     <div className="tiles-container">
       <StyledTiles>
         {
-          tiles.map((tile: any) => <TileItem tile={tile} key={tile.id} />)
+          tiles.map((tile: TileInterface) => <TileItem tile={tile} key={tile.id} />)
```
Comment (drews-believer) [inline src/components/App/ContentWrapper/HomePage/Tiles/Tiles.tsx:31]: gj

**Review by drews-believer — APPROVED** [review]
not bad)

File: src/components/App/ContentWrapper/HomePage/Tiles/Tiles.tsx (hunk 1 above)
Comment (letrylesha) [reply src/components/App/ContentWrapper/HomePage/Tiles/Tiles.tsx:31]: Thanks bro

**Review by CBond05 — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -14,6 +14,11 @@
     "@types/react-dom": "16.9.4",
     "@types/react-redux": "^7.1.5",
     "@types/react-router-dom": "^5.1.3",
+    "chart.js": "^2.9.3",
+    "chartjs": "^0.3.24",
```
Comment (didevgen) [inline package.json:18]: ?)

File: src/components/App/App.css [hunk 3]
```diff
@@ -12,4 +12,14 @@ body {
...
+}
+
+h3 {
+  margin: 0;
+  padding: 0;
+}
+h4 {
+  margin: 0;
+  padding: 0;
+}
```
Comment (didevgen) [inline src/components/App/App.css:24]: h3, h4

File: src/components/App/ContentWrapper/Chart/Chart.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  const [chartData] = React.useState(dataChart);
+
+  const getChartData = () => {
+    const data = chartData.data;
+    return data;
+  }
+  
+  return (
+    <div style={{ position: "relative", height: "350px", width: "90%", margin: '0 auto' }}>
```
Comment (didevgen) [inline src/components/App/ContentWrapper/Chart/Chart.tsx:34]: use styled component

File: src/components/App/ContentWrapper/ProfilePage/ProfileAvatar/ProfileAvatar.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    name, 
+    group, 
+    changeProfileSummaryWindow, 
+    windowStatus } = props;
+    
+  return (
+    <ProfileAvatarContainer>
+      <ProfileTitleContainer>Profile Avatar</ProfileTitleContainer>
+      <ProfileAvatarImage 
+        src="https://media-exp2.licdn.com/dms/image/C4D03AQHrET7kILmAXg/profile-displayphoto-shrink_200_200/0?e=1584576000&v=beta&t=6YQmC2IJ9grgB9FICDQvvlCYvRgyzErzsGH2jph3fKM"
```
Comment (didevgen) [inline src/components/App/ContentWrapper/ProfilePage/ProfileAvatar/ProfileAvatar.tsx:31]: move to the constant

File: src/components/App/ContentWrapper/ProfilePage/ProfileInfo/ProfileInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    <div>
+      <ProfileSummaryItem>
+        1. Name:
+        <ProfileSummaryInnerItem>
+        { profileData.name }
+        </ProfileSummaryInnerItem>
+      </ProfileSummaryItem>
+
+      <ProfileSummaryItem>
+        2. Group:
```
Comment (didevgen) [inline src/components/App/ContentWrapper/ProfilePage/ProfileInfo/ProfileInfo.tsx:37]: use spans around text elements

File: src/components/App/ContentWrapper/ProfilePage/ProfilePage.tsx [hunk 7]
```diff
@@ -1,12 +1,61 @@
...
+  profileData: any
+}
+
+const ProfilePage: React.FC<Props> = ({ getData, profileData }) => {
+  
+  
+  React.useEffect(() => {
+    getData();
+    // eslint-disable-next-line
+  }, []);
```
Comment (didevgen) [inline src/components/App/ContentWrapper/ProfilePage/ProfilePage.tsx:26]: just useEffect

File: src/components/App/ContentWrapper/ProfilePage/ProfilePage.tsx [hunk 8]
```diff
@@ -1,12 +1,61 @@
...
+  
+  if (!profileData) {
+    return <div>Loading</div>
```
Comment (didevgen) [inline src/components/App/ContentWrapper/ProfilePage/ProfilePage.tsx:29]: maybe spinner is better?

File: src/components/App/ContentWrapper/ProfilePage/ProfilePage.tsx [hunk 9]
```diff
@@ -1,12 +1,61 @@
...
+
+      <ProfileAvatar name={profileData.name} group={profileData.group}/>
+
+      <ProfileInfo profileData={profileData} />
+
+      <ProfileProgressContainer>
+        <ProfileTitleContainer>Profile Progress</ProfileTitleContainer>
+        <div>
+          <Chart />
+        </div>
```
Comment (didevgen) [inline src/components/App/ContentWrapper/ProfilePage/ProfilePage.tsx:46]: don't see the reason to wrap with div

File: src/store/profile/ProfileReducer.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+      return {
+        ...state,
+        windowStatus: !state.windowStatus
+      }
+    default:
+      return state;
+  }
+};
+
+export default profileReducer;
```
Comment (didevgen) [inline src/store/profile/ProfileReducer.ts:25]: don't see reflection in unit tests (applicable to all another sagas)

**Review by didevgen — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (letrylesha) [reply package.json:18]: it is new feature))0

File: src/components/App/ContentWrapper/ProfilePage/ProfileAvatar/ProfileAvatar.tsx (hunk 5 above)
Comment (letrylesha) [reply src/components/App/ContentWrapper/ProfilePage/ProfileAvatar/ProfileAvatar.tsx:31]: its temporarily path

**Pull request closed without merging**